lib/bolt/config/transport.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+require 'bolt/util'
+
+module Bolt
+  class Config
+    class Transport
+      attr_reader :config
+
+      def initialize(data = {}, boltdir = nil)
+        assert_hash(data)
+        @config = Bolt::Util.deep_merge(defaults, filter(data))
+        @boltdir = boltdir
+        validate
+      end
+
+      def [](key)
+        @config[key]
+      end
+
+      def to_h
+        @config
+      end
+
+      def fetch(*args)
+        @config.fetch(*args)
+      end
+
+      def include?(args)
+        @config.include?(args)
+      end
```

Comment:
This is used by a handful of spec tests. Similar to `#fetch` above, the tests call `target.options.include?`. I could either update `#options` to return the config hash or just call `targets.options.to_h.include?`.